subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/SkipEmptySourceFilesTaskExecuter.java

```diff
@@ -101,9 +112,10 @@ public TaskExecuterResult execute(TaskInternal task, TaskStateInternal state, fi
                 } else {
                     state.setOutcome(TaskExecutionOutcome.NO_SOURCE);
                 }
-                taskArtifactState.snapshotAfterTaskExecution(context);
             }
+            // TODO:lptr can we remove this?
```

Comment:
Yeah, but if you follow the code, we are always passing in an empty `FileCollection` here. Still, my earlier attempts at removing it ended up with some mysterious failures.

Reply:
Source files is a live file collection isn't it? So it's value can change. Mostly it will be a file tree for some directory, which will have files as soon as there are some files in the directory.

Reply:
I'm confused. All we do in `DefaultTaskInputsListener.onExecute()` is that we call `FileCollectionInternal.registerWatchPoints()`. And what that method does is it eagerly adds every `File` in the collection to a builder. There's nothing lazy/live about this: we take what files there are in the collection right now, which should be none at this point in the code anyway.

Reply:
If the collection is a file tree, then `getFiles()` will return an empty list for an empty directory. Registering the watch point will end up [here](https://github.com/gradle/gradle/blob/b16479080a8864f91d15540d41bc09bdbe2c0af3/subprojects/files/src/main/java/org/gradle/api/internal/file/FileSystemSubset.java#L134) where we add the root directory of the file tree as a watch point.

Reply:
Yeah, see my comment just above.

Reply:
Makes sense, thanks, removed the comment.